src/app/clusters/joint-fabric-administrator-server/joint-fabric-administrator-server.cpp

```diff
@@ -123,10 +152,18 @@ bool emberAfJointFabricAdministratorClusterAddICACCallback(
 
     ChipLogProgress(Zcl, "JointFabricAdministrator: Received a AddICAC command");
 
+    auto nonDefaultStatus = Status::Success;
+
+    auto & failSafeContext = Server::GetInstance().GetFailSafeContext();
+    VerifyOrExit(failSafeContext.IsFailSafeArmed(commandObj->GetAccessingFabricIndex()),
+                 nonDefaultStatus = Status::FailsafeRequired);
+    VerifyOrExit(failSafeContext.IcacCsrRequestHasBeenInvoked(), nonDefaultStatus = Status::ConstraintError);
```

Comment:
This check and status is not in the spec.  The logic in the spec there seems quite different in general?

Reply:
I don't see this check in the spec, even with https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/11848/files applied.

Reply:
That's right, this check (`failSafeContext.IcacCsrRequestHasBeenInvoked()`) is not present in the spec.:
-  initial version of this PR was doing this check but this was not following the spec. (as you highlighted)
- the current version  of the PR has this check removed